mathesar/api/serializers/columns.py

```diff
@@ -147,15 +151,19 @@ class Meta(SimpleColumnSerializer.Meta):
     def validate(self, data):
         data = super().validate(data)
         # Reevaluate column display options based on the new column type.
-        if TYPE_KEY in data and DISPLAY_OPTIONS_KEY not in data:
-            if self.instance:
+        if TYPE_KEY in data and self.instance:
+            target_types = self.get_valid_target_types(self.instance)
+            if data[TYPE_KEY].lower() not in target_types:
```

Comment:
For this check, it would be better to use the enum version of the type. Please see 
```python
db.types.operations.convert.get_db_type_enum_from_id
```
You can use that to convert the given type ID to the appropriate Enum value, then check for membership in `self.instance.valid_target_types` directly.